packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasEstimates.ts

```diff
@@ -0,0 +1,93 @@
+import { BigNumber } from 'ethers'
+import { Provider } from '@ethersproject/providers'
+import useSWR from 'swr'
+
+import { GasEstimates } from '../arbTokenBridge.types'
+import { depositTokenEstimateGas } from '../../util/TokenDepositUtils'
+import { withdrawInitTxEstimateGas } from '../../util/WithdrawalUtils'
+import { Address } from '../../util/AddressUtils'
+import { depositEthEstimateGas } from '../../util/EthDepositUtils'
+
+type TransferType = 'deposit' | 'withdrawal'
+
+async function fetcher([
+  walletAddress,
+  txType,
+  parentChainProvider,
+  childChainProvider,
+  tokenParentChainAddress,
+  amount
+]: [
+  walletAddress: Address,
+  txType: TransferType,
+  parentChainProvider: Provider | undefined,
+  childChainProvider: Provider,
```

Comment:
if we did source/destination we then don't need txType?

Reply:
we can do an internal check to see if we should estimate deposit or withdrawal gas

Reply:
gas estimation for deposits use both parent chain and child chain providers
gas estimation for withdrawals only use child chain provider

so I am not sure how using source/destination provider would help?